docs/design.md

```diff
@@ -0,0 +1,67 @@
+# Design
+
+## Models
+
+A Model object represents a logical ML model.
+
+The object describes where the source code for the model should come from. This code will be stored in a container image. That container image will initially be run in order to store the initial weights and biases in a bucket that is mounted into the container as a directory. This is most useful in cases where the Model is representing a pre-trained model.
+
+The Model object can optionally describe the training parameters including a reference to the Dataset that should be used for training. In this case, the Model object represents the eventual output of a training/finetuning operation.
+
+```yaml
+kind: Model
+spec:
+  source: {} # Required
```

Comment:
could we make source optional such that you can create a notebook with an empty model attached?

Reply:
That would also allow us to have a FinetuningJob CRD later if needed